ValidationLibrary.AzureFunctions.Tests/StatusEndpointTests.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using NSubstitute;
+using NUnit.Framework;
+using ValidationLibrary.Rules;
+
+namespace ValidationLibrary.AzureFunctions.Tests
+{
+    [TestFixture]
+    public class StatusEndpointTests
+    {
+        private StatusEndpoint _statusEndpoint;
+
+        [Test]
+        public void Run_NormalStatusEndpointCheck()
+        {
+            Environment.SetEnvironmentVariable("GitHub:Organization", "mock");
+            Environment.SetEnvironmentVariable("GitHub:Token", "mock");
+
+            IHost host = new HostBuilder().ConfigureWebJobs(new Startup().Configure).Build();
```

Comment:
We don't want to test our host building here, host startup testing (DI building etc.) should be testesd in StartupTests. There we should just tests that we can create StatusEndpoint.

You should copy the implementation (change the relevant parts of course) from `ValidationLibrary.AzureFunctions.Tests.RepositoryValidatorTests`

Reply:
Hmm, yes I'm not so sure about all the substituting.For. Do you mean like this?

Reply:
No quite. Because we are writing an unit test for `StatusEndpoint`:

1. We don't want any `Environment` stuff, because `StatusEndpoint` doesn't use those.
1. We don't want to create actual instances of `IValidationRule`, those are something that we want to mock, because we are not testing rules.

These are just a couple of read flags that will help us in future to see when we are breaking layers.

I'll write more specific comments to substitute usage in the code.